server/server.go

```diff
@@ -19,11 +29,66 @@ func New(cfg *config.Config) (*Server, error) {
 		cfg: *cfg,
 	}
 
+	if err := s.Setup(); err != nil {
+		return nil, err
+	}
+
 	router := chi.NewRouter()
+	router.NotFound(NotFound)
+	router.Route("/photos", func(router chi.Router) {
+		router.Get("/", s.GetPhotos)
+		router.Post("/", s.PostPhoto)
+		router.Get("/ids", s.GetPhotoIDs)
+		router.Route("/{pid}", func(router chi.Router) {
+			router.Use(s.PhotoCtx)
+			router.Get("/image.jpg", s.GetPhoto)
+			router.Get("/thumb.jpg", s.GetThumbnail)
+			router.Get("/meta", s.GetPhotoMetadata)
+		})
+	})
+
 	s.handler = router
 	return s, nil
 }
 
+func (s *Server) Setup() error {
+	if !CanAccessDirectory(s) {
+		return errors.New(fmt.Sprintf("Directory %s not accessible", s.cfg.PhotosDirectory))
+	}
+
+	if _, err := os.Stat(s.cfg.ImgFolder()); err != nil {
+		os.Mkdir(s.cfg.ImgFolder(), 0775)
+	}
+
+	if _, err := os.Stat(s.cfg.ConfFolder()); err != nil {
+		os.Mkdir(s.cfg.ConfFolder(), 0775)
+	}
+
+	db, err := storm.Open(s.cfg.StormFile())
+	if err != nil {
+		return err
+	}
+	s.db = db
+
+	if err := s.db.Init(&Photo{}); err != nil {
+		return err
+	}
+
+	exif.RegisterParsers(mknote.All...)
+
+	return nil
+}
+
+func CanAccessDirectory(serv *Server) bool {
+	stat, statErr := os.Stat(serv.cfg.PhotosDirectory)
+	if statErr != nil || !stat.IsDir() {
+		return false
+	}
+	readErr := unix.Access(serv.cfg.PhotosDirectory, unix.R_OK)
+	writeErr := unix.Access(serv.cfg.PhotosDirectory, unix.W_OK)
```

Comment:
Can this be done with `os.Stat()`? So we're not relying on OS-specific stuff. (I may want to run Hotshots on my Windows XP SP2 box.)

Reply:
Dear god